.github/workflows/dependency_changelog.yml

```diff
@@ -0,0 +1,111 @@
+# **what?**
+# When dependabot create a PR, it always adds the `dependencies` label. This
+# action will add a corresponding changie yaml file to that PR when that label is added.
+# The file is created off a template:
+#
+# kind: Dependencies
+# body: <PR title>
+# time: <current timestamp>
+# custom:
+#   Author: dependabot
+#   Issue: 4904
+#   PR: <PR number>
+#
+# **why?**
+# Automate changelog generation for more visability with automated dependency updates via dependabot.
+
+# **when?**
+# Once a PR is created and it has been correctly labeled with `dependencies`.  The intended use
+# is for the PRs created by dependabot.  You can also manually trigger this by adding the
+# `dependencies` label at any time.
+
+name: Dependency Changelog
+
+on:
+  pull_request:
+    # catch when the PR is opened with the label or when the label is added
+    types: [opened, labeled]
+
+permissions:
+  contents: write
+  pull-requests: read
```

Comment:
Since it's committing to the branch I think this permission setting is least privilege.

Reply:
I think this is correct